astrodb_utils/spectra.py

```diff
@@ -0,0 +1,169 @@
+import importlib.util
+import logging
+import sys
+
+import astropy.units as u
+import numpy as np
+from specutils import Spectrum1D
+
+from astrodb_utils import AstroDBError
+
+matplotlib_check = importlib.util.find_spec("matplotlib")
+if matplotlib_check is not None:
+    import matplotlib.pyplot as plt
+
+
+__all__ = [
+    "check_spectrum_class", 
+    "check_spectrum_not_nans", 
+    "check_spectrum_wave_units", 
+    "check_spectrum_flux_units", 
+    "check_spectrum_plottable"
+    ]
+
+logger = logging.getLogger(__name__)
+
+
+def check_spectrum_class(spectrum, raise_error=True):
+    try:
+        Spectrum1D.read(spectrum)
+        return True
+    except Exception as error_message:
+        msg = f"Unable to load file as Spectrum1D object:{spectrum}"
+        logger.debug(f"{error_message}")
+        if raise_error:
+            logger.error(msg)
+            raise AstroDBError(msg)
+        else:
+            logger.warning(msg)
+            return False
+
+
+def check_spectrum_not_nans(spectrum, raise_error=True):
+    nan_check: np.ndarray = ~np.isnan(spectrum.flux) & ~np.isnan(spectrum.spectral_axis)
+    wave = spectrum.spectral_axis[nan_check]
+    if not len(wave):
+        msg = "Spectrum is all NaNs"
+        if raise_error:
+            logger.error(msg)
+            raise AstroDBError(msg)
+        else:
+            logger.warning(msg)
+            return False
+    else:
+        return True
+
+
+def check_spectrum_wave_units(spectrum, raise_error=True):
+    try:
+        spectrum.spectral_axis.to(u.micron).value
+        return True
+    except AttributeError as e:
+        logger.debug(f"{e}")
+        msg = f"Unable to parse spectral axis: {spectrum}"
+        if raise_error:
+            logger.error(msg)
+            raise AstroDBError(msg)
+        else:
+            logger.warning(msg)
+            return False
+    except u.UnitConversionError as e:
+        logger.debug(f"{e}")
+        msg = f"Unable to convert spectral axis to microns:  {spectrum}"
+        if raise_error:
+            logger.error(msg)
+            raise AstroDBError(msg)
+        else:
+            logger.warning(msg)
+            return False
+    except ValueError as e:
+        logger.debug(f"{e}")
+        msg = f"Value error: {spectrum}:"
+        if raise_error:
+            logger.error(msg)
+            raise AstroDBError(msg)
+        else:
+            logger.warning(msg)
+            return False
+
+
+def check_spectrum_flux_units(spectrum, raise_error=True):
+    try:
```

Comment:
yeah, under the hood this could also maybe be function'd out, because it does the same operation as checking the units in the wavelength axis.

Reply:
I also agree but I wasn't sure how to make it one function since the feedback to the user is quite different in the two cases. Again, advice is welcome.